environment.py

```diff
@@ -0,0 +1,150 @@
+    output_result(create_title_string("ISOMORPHISMS"))
+    iso_start = time.time()
+    preprocessed_graphs = preprocess_isomorphisms(graphs)
+    isomorphs = calculate_isomorphisms(preprocessed_graphs)
+    iso_time = time.time() - iso_start
+
+    output_result(create_title_string("AUTOMORPHISMS"))
+    graphs = [graphs[0] for graphs in isomorphs]
+    auto_start = time.time()
+    preprocessed_graphs = preprocess_automorphisms(graphs)
+    automorphs = calculate_automorphisms(preprocessed_graphs)
+    auto_time = time.time() - auto_start
+
+    return isomorphs, iso_time, automorphs, auto_time
+
+
+def preprocess_isomorphisms(graphs: List[Graph]) -> List[Graph]:
+    return graphs
+
+
+def calculate_isomorphisms(graphs: List[Graph]) -> List[List[Graph]]:
+    isomorphs = [[]]
+    isomorphs[0] = [graphs.pop(0)]
+    for i in range(len(graphs)):
+        added = False
+        start_time = time.time()
+        for j in range(len(isomorphs)):
+            if is_isomorphisms(isomorphs[j][0], graphs[i]):
+                isomorphs[j].append(graphs[i])
+                added = True
+                output_result(graphs[i].name + " and " + isomorphs[j][0].name + " are isomorphisms (" + str(
+                    time.time() - start_time) + ")")
+                break
+        if not added:
+            isomorphs.append([graphs[i]])
+            output_result(graphs[i].name + " has no isomorphisms yet (" + str(time.time() - start_time) + ")")
+    return isomorphs
+
+
+def preprocess_automorphisms(graphs: List[Graph]) -> List[Graph]:
+    return graphs
+
+
+def calculate_automorphisms(graphs: List[Graph]) -> List[int]:
+    automorphisms = []
+    for graph in graphs:
+        start_time = time.time()
+        num_automorphisms = get_number_automorphisms(graph)
+
+        output_result(graph.name + "\'s group has " + str(num_automorphisms) + " automorphisms (" + str(
+            time.time() - start_time) + ")")
+        automorphisms.append(num_automorphisms)
+    return automorphisms
+
+
+def stringify_result(file: str, isomorphs: List[List[Graph]], iso_time: float, automorphs: List[int],
+                     auto_time: float) -> str:
+    title_string = create_title_string(get_file_title(file))
+    data_string = create_data_string(isomorphs, automorphs)
+    footer_string = create_footer_string(iso_time, auto_time)
+    return title_string + '\n' + data_string + '\n' + footer_string + '\n'
+
+
+def create_title_string(title: str) -> str:
+    return '\n' + title + '\n' + '{:_<41}'.format('') + '\n'
+
+
+def get_file_title(file: str) -> str:
+    return file.rsplit(os.path.sep, 1)[1]
+
+
+def create_data_string(isomorphs: List[List[Graph]], automorphs: List[int]) -> str:
+    output_format = '{:<20} {:>20}'
+    data_string = output_format.format('ISOMORPHISMS', 'AUTOMORPHISMS')
+    for m in range(len(isomorphs)):
+        iso_string = str(len(isomorphs[m])) + ' ('
```

Comment:
Je kunt ook het volgende doen
`iso_string = str(len(isomorphs[m])) + ' ( ' + ', '.join [graph.name for graph in isomorphs[m]] + ')'